Linkta-org/core
# Setup Integration Testing

## Description
- Installed additional testing dependencies
- Refactored Vitest config to separate unit and integration tests
- Updated test scripts in package.json
- Updated existing unit tests(fixed failed App.unit.test in `/client` && replaced the placeholder in `/server` with a real test)
- Added sample integration tests
- Add CI workflows for unit & integration tests

**Resources:**
https://www.synthesized.io/post/balancing-happy-path-negative-testing
https://www.initialyze.com/insights/the-importance-of-integration-testing
https://kentcdodds.com/blog/write-tests

Fixes #92 

## Type of change

## How Can this be tested? Testing Plan to review this PR

Manual testing:
```bash
cd client
npm run test:unit
npm run test:integration
npm run test
```
```bash
cd server
npm run test:unit
npm run test:integration
npm run test
```

Expected Result: All tests should pass successfully
![image](https://github.com/user-attachments/assets/e170ebb1-8bde-45ba-adc1-bab452733a08)
![image](https://github.com/user-attachments/assets/d178f04a-c48e-4791-bb45-7db3780f6902)
![image](https://github.com/user-attachments/assets/46a6e2e2-e799-493b-8855-5147683833ca)
![image](https://github.com/user-attachments/assets/acf35a4d-bc3a-467e-97ef-254fac2b1c73)

Automated testing through Github Actions (all passed)
![image](https://github.com/user-attachments/assets/2349039c-48bf-4f48-97e1-865a24a88e36)

## Checklist:



- [X] I have reviewed the [**Security Best Practices Document**](../docs/guides/development/Security_Best_Practices_Guide.md).
- [X] I have followed the [**Naming Conventions Guide**](./docs/guides/development/Naming_Conventions_Guide.md) for my changes.
